runtime/android/core_internal/src/org/xwalk/core/internal/XWalkMediaPlayerResourceLoadingFilter.java

```diff
@@ -39,10 +45,20 @@ public boolean shouldOverrideResourceLoading(MediaPlayer mediaPlayer,
                     context.getAssets().openFd(AndroidProtocolHandler.getAssetPath(uri));
             mediaPlayer.setDataSource(
                     afd.getFileDescriptor(), afd.getStartOffset(), afd.getLength());
-
             return true;
         } catch (Exception e) {
             return false;
         }
     }
+
+    @Override
+    public ExternalMediaPlayer getExternalMediaPlayer() {
+        ExternalMediaPlayer exMediaPlayer = mContentsClientBridge.getExternalMediaPlayer();
+
+        if (exMediaPlayer == null) {
+            exMediaPlayer = new ExternalMediaPlayer();
```

Comment:
> If the media player in XWalkContentsClientBridge is null, everytime you call getExternalMediaPlayer from XWalkMediaPlayerResourceLoadingFilter, a new media player will be created. Is that alright?

`getExternalMediaPlayer` is used as below, so it's not created again.
but there is still some confusedness for reading code, so i will move it to getLocalPlayer
> 
protected ExternalMediaPlayer getLocalPlayer() {
        if (mPlayer == null) {
            mPlayer = sResourceLoadFilter.getExternalMediaPlayer();
        }
        return mPlayer;
    } 

The refined result is
> 
protected ExternalMediaPlayer getLocalPlayer() {
        if (mPlayer == null) {
            mPlayer = sResourceLoadFilter.getExternalMediaPlayer();
            if (mPlayer == null) mPlayer = new ExternalMediaPlayer();
        }
        return mPlayer;
    }